test-app/src/main/java/net/rubygrapefruit/platform/test/Main.java

```diff
@@ -366,25 +368,38 @@ private static void fileSystems() {
         System.out.println("* File systems: ");
         for (FileSystemInfo fileSystem : fileSystems.getFileSystems()) {
             System.out.println(String.format("    * %s -> %s (type: %s %s, case sensitive: %s, case preserving: %s)",
-                    fileSystem.getMountPoint(), fileSystem.getDeviceName(), fileSystem.getFileSystemType(),
-                    fileSystem.isRemote() ? "remote" : "local", fileSystem.isCaseSensitive(), fileSystem.isCasePreserving()));
+                fileSystem.getMountPoint(), fileSystem.getDeviceName(), fileSystem.getFileSystemType(),
+                fileSystem.isRemote() ? "remote" : "local", fileSystem.isCaseSensitive(), fileSystem.isCasePreserving()));
         }
 
         System.out.println();
     }
 
     private static void watch(String path) {
-        FileWatcher watcher = createWatcher(path, new FileWatcherCallback() {
+        final BlockingQueue<FileWatchEvent> eventQueue = new ArrayBlockingQueue<FileWatchEvent>(16);
+        Thread processorThread = new Thread(new Runnable() {
             @Override
-            public void pathChanged(Type type, String changedPath) {
-                System.out.printf("Change detected: %s / '%s'%n", type, changedPath);
-            }
-
-            @Override
-            public void reportError(Throwable ex) {
-                ex.printStackTrace();
+            public void run() {
+                while (!Thread.currentThread().isInterrupted()) {
+                    FileWatchEvent event;
+                    try {
+                        event = eventQueue.take();
+                    } catch (InterruptedException e) {
+                        break;
```

Comment:
The thread should clear the interrupted flag here, right?

Reply:
That'd be prudent, yes. We could also join the thread. But it's just a daemon thread and we are about to exit the application anyway, so I didn't bother. Do you think it's important?

Reply:
It just catched my eye.